core/src/main/scala/io/qbeast/spark/writer/Rollup.scala

```diff
@@ -78,6 +78,37 @@ private[writer] class Rollup(limit: Double) {
     }.toMap
   }
 
+  private def areSiblings(cube_a: CubeId, cube_b: CubeId): Boolean = {
+    val sameParent = cube_a.parent == cube_b.parent
+    val differentCube = cube_a != cube_b
+    sameParent && differentCube
+  }
+
+  /*
+   * Ordering for cube identifiers. The cube identifiers are ordered by their depth in ascending
+   * order. If the depth is the same then the cube identifiers are ordered by in reverse order.
+   * This ordering is used in the priority queue to process the cube identifiers in the correct
+   * order, i.e., from the deepest to the shallowest, and from the leftmost to the rightmost:
+   *    0                     root
+   *    1           c0                   c1
+   *    2     c00         c01       c10         c11
+   * The priority queue will process the cube identifiers in the following order:
+   * c00, c01, c10, c11, c0, c1, root.
+   * c00 -> c01 -> c0, c10 -> c11 -> c1, c0 -> c1 -> root
+   */
+  private object CubeIdOrdering extends Ordering[CubeId] {
+
+    override def compare(x: CubeId, y: CubeId): Int = {
```

Comment:
Can you add a few tests that show how this order is different from the default CubeId ordering?

Reply:
Done